immer/detail/util.hpp

```diff
@@ -232,16 +232,16 @@ auto static_if(F&& f) -> std::enable_if_t<b>
     std::forward<F>(f)(empty_t{});
 }
 template <bool b, typename F>
-auto static_if(F&& f) -> std::enable_if_t<!b>
```

Comment:
Maybe these should just be annotated as maybe unused instead of removed

Reply:
Sorry for the very late review! This changes the signature of the function so I don't think this work. Maybe you can do it like this:
```c++
template <bool b, typename F>
auto static_if(F&&) -> std::enable_if_t<!b>
```
??